docs/contributing/MODEL_MANAGER.md

```diff
@@ -0,0 +1,1214 @@
+
+** TO DO: ** Change the logic so that the caller's metadata fields take
+precedence over those provided by the source.
+
+
+#### Other installer methods
+
+This section describes additional, less-frequently-used attributes and
+methods provided by the installer class.
+
+##### installer.wait_for_installs()
+
+This is equivalent to the `DownloadQueue` `join()` method. It will
+block until all the active jobs in the install queue have reached a
+terminal state (completed, errored or cancelled).
+
+##### installer.queue, installer.store, installer.config
+
+These attributes provide access to the `DownloadQueueServiceBase`,
+`ModelConfigRecordServiceBase`, and `InvokeAIAppConfig` objects that
+the installer uses.
+
+For example, to temporarily pause all pending installations, you can
+do this:
+
+```
+installer.queue.pause_all_jobs()
+```
+##### key = installer.register_path(model_path, overrides), key = installer.install_path(model_path, overrides)
+
+These methods bypass the download queue and directly register or
+install the model at the indicated path, returning the unique ID for
+the installed model.
+
+Both methods accept a Path object corresponding to a checkpoint or
+diffusers folder, and an optional dict of attributes to use to
+override the values derived from model probing.
+
+The difference between `register_path()` and `install_path()` is that
+the former will not move the model from its current position, while
+the latter will move it into the `models_dir` hierarchy.
+
+##### installer.unregister(key)
+
+This will remove the model config record for the model at key, and is
+equivalent to `installer.store.unregister(key)`
+
+##### installer.delete(key)
+
+This is similar to `unregister()` but has the additional effect of
+deleting the underlying model file(s) -- even if they were outside the
+`models_dir` directory!
+
+##### installer.conditionally_delete(key)
+
+This method will call `unregister()` if the model identified by `key`
+is outside the `models_dir` hierarchy, and call `delete()` if the
+model is inside.
+
+#### List[str]=installer.scan_directory(scan_dir: Path, install: bool)
+
+This method will recursively scan the directory indicated in
+`scan_dir` for new models and either install them in the models
+directory or register them in place, depending on the setting of
+`install` (default False).
+
+The return value is the list of keys of the new installed/registered
+models.
+
+#### installer.scan_models_directory()
+
+This method scans the models directory for new models and registers
+them in place. Models that are present in the
+`ModelConfigRecordService` database whose paths are not found will be
+unregistered.
+
+#### installer.sync_to_config()
+
+This method synchronizes models in the models directory and autoimport
+directory to those in the `ModelConfigRecordService` database. New
+models are registered and orphan models are unregistered.
+
+#### hash=installer.hash(model_path)
```

Comment:
Is this a member of the installer for any reason? Or should it just be a standalone utility?

Reply:
Just a convenience. There's a standalone class in the backend called `FastModelHash` that does the actual work. I'll just remove this from the code and docs.